src/common/log.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import log, {LevelOption} from 'electron-log';
+
+import Util from 'common/utils/util';
+
+export const setLoggingLevel = (level: string) => {
+    if (log.transports.file.level === level) {
+        return;
+    }
+    log.error('Logger', 'Log level set to:', level);
+
+    log.transports.console.level = level as LevelOption;
+    log.transports.file.level = level as LevelOption;
+};
+
+export class Logger {
+    private prefixes: string[];
+
+    constructor(...prefixes: string[]) {
+        this.prefixes = prefixes;
+    }
+
+    withPrefix = (...prefixes: string[]) => {
+        return {
+            error: this.prefixed(log.error, ...prefixes),
+            warn: this.prefixed(log.warn, ...prefixes),
+            info: this.prefixed(log.info, ...prefixes),
+            verbose: this.prefixed(log.verbose, ...prefixes),
+            debug: this.prefixed(log.debug, ...prefixes),
+            silly: this.prefixed(log.silly, ...prefixes),
+            log: this.prefixed(log.log, ...prefixes),
+        };
+    }
```

Comment:
So in 90% of cases, most modules will just call `new Logger(...some prefixes)` and then will always use the object returned, avoiding the need to use `withPrefix`. But I'm providing this here for the more dynamic cases like the `WebContentsEventManager` where we don't know the `webContentsId` immediately.

You may think, "well that's a code smell" and you're right. But it's coming from the WebContentsEventsManager, which also should be refactored because there is a way to know the id ahead of time and avoid using this.

tl;dr It's there for convenience right now, should probably be removed when we fix other things :P